src/text/parsers/decimal.rs

```diff
@@ -91,13 +88,21 @@ fn decimal_from_text_components(
     // we've just removed the underscores above, the `from_str` methods below should always
     // succeed.
     let magnitude: Magnitude = if magnitude_text.len() < MAX_U64_DIGITS {
-        u64::from_str(&magnitude_text)
-            .expect("parsing coefficient magnitude as u64 failed")
-            .into()
+        let value = u64::from_str(&magnitude_text)
+            .or_fatal_parse_error(
+                input,
+                "parsing coefficient magnitude as u64 failed",
+            )?
```

Comment:
🗺️ Bubble up information about the parse failure instead of panicking. `or_fatal_parse_error` is provided by the extension trait `OrFatalParseError` and is defined for all configurations of `std::result::Result`.